frontend/src/components/forms/CreateLog.tsx

```diff
@@ -0,0 +1,394 @@
+
+// Ideally we should be storing this information in the database
+const BUILDINGS = [
+    { label: "144", value: "144 Erb St. West" },
+    { label: "362", value: "362 Erb St. West" },
+    { label: "402", value: "402 Erb St. West" },
+]
+
+// Replace this with the tags from the db once the API and table are made
+const TAGS = [
+    { label: "Tag A", value: "A" },
+    { label: "Tag B", value: "B" },
+    { label: "Tag C", value: "C" },
+]
+
+// Replace this with the residents from the db
+const RESIDENTS = [
+    { label: "DE307", value: "DE307" },
+    { label: "AH206", value: "AH206" },
+    { label: "MB404", value: "MB404" },
+]
+
+// Replace this with the users from the db
+const EMPLOYEES = [
+    { label: "Huseyin", value: "Huseyin" },
+    { label: "John Doe", value: "John Doe" },
+]
+
+// Changes the border of the Select components if the input is invalid
+function getBorderStyle(state: any, error: boolean): string {
+    if (state.isFocused) {
+        return "2px solid #3182ce";
+    }
+    if (error) {
+        return "2px solid #e53e3e";
+    }
+
+    return "1px solid #cbd5e0";
+}
+
+const CreateLog = () => {
+    const [employee, setEmployee] = useState("Huseyin"); // currently, the select for employees is locked and should default to current user. Need to check if admins/regular staff are allowed to change this
+    const [date, setDate] = useState(new Date());
+    const [time, setTime] = useState(date.toLocaleTimeString([], { hour: '2-digit', minute: '2-digit', hour12: false }));
+    const [building, setBuilding] = useState("");
+    const [resident, setResident] = useState("");
+    const [tags, setTags] = useState<string[]>([]);
+    const [attnTo, setAttnTo] = useState("");
+    const [notes, setNotes] = useState("");
+    const [flagged, setFlagged] = useState(false);
+
+    // error states for non-nullable inputs
+    const [employeeError, setEmployeeError] = useState(false);
+    const [dateError, setDateError] = useState(false);
+    const [timeError, setTimeError] = useState(false);
+    const [buildingError, setBuildingError] = useState(false);
+    const [residentError, setResidentError] = useState(false);
+    const [notesError, setNotesError] = useState(false);
+
+    // if we need functionality to change the selected employee, handle should go here
+
+    const handleDateChange = (newDate: Date) => {
+        setDate(newDate);
+    };
+
+    const handleTimeChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setTime(e.target.value);
+        const timeRegex = /^(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$/; // Regex to match time format HH:mm
+
+        // Check to see if the input is valid, prevents an application crash
+        if (timeRegex.test(e.target.value)) {
+            const [hour, minute] = e.target.value.split(":");
+            const updatedDate = new Date(date); // update the time values of the current date state
+            updatedDate.setHours(parseInt(hour, 10));
```

Comment:
Curious as to why we're also changing hours/minutes here? Shouldn't the time always correspond to the date picked, changing it shouldn't affect the date no?

Reply:
Whenever the date state is updated, the dd/mm/yyyy values change but never the hh:mm value. This updates both the time state, so that it renders correctly, and the date state, so that when it is submitted, it'll be at the correct time. In our db, we store the date value in one column (instead of date and time).